### PR TITLE
sane_backends: bump version

### PR DESCRIPTION
**Pull request by kenmays**
Built sane_backends 1.0.29 successfully on hrev54109 x86 and hrev54139 x86_64.. 
All unit tests passed. Scanner used: Fujitsu ScanSnap iX1500
Closes: #4903

### CODE REVIEW

**Comment by Vidrep** [comment]
Builds, installs and tested working with both Canon and Epson multifunction printer/scanner

File: media-gfx/sane_backends/sane_backends-1.0.29.recipe [hunk 1]
```diff
@@ -13,16 +13,15 @@ backends are welcome, too, however).
 This package includes scanners backends, the command line frontend scanimage, \
 the saned server and the sane-find-scanner utility, along with their \
 documentation."
+
```
Comment (Begasus) [inline media-gfx/sane_backends/sane_backends-1.0.29.recipe:16]: No new line between DESCRIPTION and HOMEPAGE

File: media-gfx/sane_backends/sane_backends-1.0.29.recipe [hunk 2]
```diff
@@ -13,16 +13,15 @@ backends are welcome, too, however).
...
 LICENSE="GNU LGPL v2"
-REVISION="6"
-SOURCE_URI="https://gitlab.com/sane-project/backends/-/archive/$portVersion/backends-$portVersion.tar.gz"
-CHECKSUM_SHA256="9f263bb8d7ea56644e71e294a50763525b6b8e3e27ee1d55bea52658a939f0c9"
-SOURCE_DIR="backends-$portVersion"
-PATCHES="
-	sane_backends-$portVersion.patchset
-	"
+REVISION="1"
+SOURCE_URI="https://gitlab.com/sane-project/backends/uploads/54f858b20a364fc35d820df935a86478/sane-backends-1.0.29.tar.gz"
```
Comment (Begasus) [inline media-gfx/sane_backends/sane_backends-1.0.29.recipe:21]: Try to use $portVersion:
```diff
- SOURCE_URI="https://gitlab.com/sane-project/backends/uploads/54f858b20a364fc35d820df935a86478/sane-backends-1.0.29.tar.gz"
+ SOURCE_URI="https://gitlab.com/sane-project/backends/uploads/54f858b20a364fc35d820df935a86478/sane-backends-$portVersion.tar.gz"
```

File: media-gfx/sane_backends/sane_backends-1.0.29.recipe [hunk 3]
```diff
@@ -78,10 +84,7 @@ BUILD_PREREQUIRES="
 
 BUILD()
 {
-	libtoolize --force --copy --install
-	aclocal
-	autoconf
-	automake --add-missing
+	export BACKENDS="abaton agfafocus apple artec artec_eplus48u as6e avision bh canon canon630u canon_dr cardscan coolscan coolscan2 coolscan3 dc25 dc210 dc240 dell1600n_net dmc epjitsu epson epson2 epsonds fujitsu genesys gt68xx hp hp3500 hp3900 hp4200 hp5400 hp5590 hpljm1005 hs2p ibm kodak kodakaio kvs1025 kvs20xx leo lexmark ma1509 magicolor matsushita microtek microtek2 mustek mustek_usb nec net niash pie pieusb pixma plustek plustek_pp ricoh ricoh2 rts8891 s9036 sceptre sharp sm3600 sm3840 sp15c st400 stv680 tamarack teco1 teco2 teco3 test u12 umax umax_pp umax1220u xerox_mfp p5 pnm"
```
Comment (waddlesplash) [inline media-gfx/sane_backends/sane_backends-1.0.29.recipe:87]: What's the point of this?

File: media-gfx/sane_backends/sane_backends-1.0.29.recipe [hunk 4]
```diff
@@ -119,3 +123,8 @@ INSTALL()
 	packageEntries devel \
 		$developDir
 }
+
+TEST()
+{
+        make check -j 4
```
Comment (waddlesplash) [inline media-gfx/sane_backends/sane_backends-1.0.29.recipe:129]: Should use $jobArgs.

File: media-gfx/sane_backends/sane_backends-1.0.29.recipe [hunk 5]
```diff
@@ -90,11 +93,12 @@ BUILD()
...
-		--with-docdir=$docDir \
+		--docdir=$docDir \
+		--with-libcurl \
 		--enable-pthread \
-		--disable-latex \
-		--disable-translations \
-		--enable-libusb_1_0 \
+		--without-api-spec \
+		--disable-locking \
+		--with-usb \
```
Comment (waddlesplash) [inline media-gfx/sane_backends/sane_backends-1.0.29.recipe:101]: Why were these arguments changed?

Comment (kenmays) [reply media-gfx/sane_backends/sane_backends-1.0.29.recipe:101]: Newer sane builds use '--with-usb' flag.

File: media-gfx/sane_backends/sane_backends-1.0.29.recipe (hunk 3 above)
Comment (kenmays) [reply media-gfx/sane_backends/sane_backends-1.0.29.recipe:87]: Some backends are specific to certain OS platforms (like Mac OS X) or don't build/test properly. This helps filter them out.

Comment (waddlesplash) [reply media-gfx/sane_backends/sane_backends-1.0.29.recipe:87]: Doesn't the configure script detect that automatically?

Comment (kenmays) [reply media-gfx/sane_backends/sane_backends-1.0.29.recipe:87]: Basically, I filtered out drivers that didn't build/test properly on Haiku. Very minor issue, but helps
isolates/filter out faulty (or unwanted) drivers under test review.

Comment (waddlesplash) [reply media-gfx/sane_backends/sane_backends-1.0.29.recipe:87]: I don't think it's worth it, we should just leave everything enabled.

Comment (kenmays) [reply media-gfx/sane_backends/sane_backends-1.0.29.recipe:87]: Won't work...some filters are not 100%. We need to filter them out.

Comment (waddlesplash) [reply media-gfx/sane_backends/sane_backends-1.0.29.recipe:87]: Can you please be more specific? Just having unit tests fail is not a reason to disable them.

Comment (kenmays) [reply media-gfx/sane_backends/sane_backends-1.0.29.recipe:87]: Fixed!!

**Comment by Begasus** [comment]
Any specific reason this was/is disabled for x86_gcc2?

**Comment by kenmays** [comment]
The GCC2 version of sane-backends 1.0.27 was not supported - so the 1.0.29 port didn't enable it either. Possibility: Port the GCC2 patchset from 1.0.25 to 1.0.29 ?!?

**Comment by kenmays** [comment]
GCC2 support dropped since 1.0.27.

**Comment by Begasus** [comment]
Thanks for reporting back, didn't fully investigated, but so far I only ran into parse errors on this version, download for 1.0.25 is lost somewhere along the line, so we probably need to change SOURCE_URI for this one

**Pull request merged**